persist-cli/src/main/java/io/ballerina/persist/introspect/Introspector.java

```diff
@@ -0,0 +1,289 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.persist.introspect;
+
+import io.ballerina.persist.inflector.CaseConverter;
+import io.ballerina.persist.inflector.Pluralizer;
+import io.ballerina.persist.introspectiondto.SQLEnum;
+import io.ballerina.persist.introspectiondto.SQLForeignKey;
+import io.ballerina.persist.introspectiondto.SQLTable;
+import io.ballerina.persist.models.Entity;
+import io.ballerina.persist.models.EntityField;
+import io.ballerina.persist.models.Enum;
+import io.ballerina.persist.models.EnumMember;
+import io.ballerina.persist.models.Index;
+import io.ballerina.persist.models.Module;
+import io.ballerina.persist.models.Relation;
+import io.ballerina.persist.models.SQLType;
+import io.ballerina.persist.utils.ScriptRunner;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.Integer.parseInt;
+
+
+
+
+/**
+ * Database Introspector class.
+ *
+ *
+ */
+public abstract class Introspector {
+
+    protected final Connection connection;
+    protected String databaseName;
+    protected abstract String getTablesQuery();
+    protected abstract String getColumnsQuery(String tableName);
+    protected abstract String getIndexesQuery(String tableName);
+    protected abstract String getForeignKeysQuery(String tableName);
+    protected abstract String getEnumsQuery();
+
+    private List<SQLTable> tables;
+
+    private List<SQLEnum> sqlEnums;
+    private final Module.Builder moduleBuilder;
+
+    private final Map<String, Entity> entityMap;
+
+    private final List<SQLForeignKey> sqlForeignKeys;
+
+    public Introspector(Connection connection, String databaseName) {
+        this.connection = connection;
+        this.databaseName = databaseName;
+        this.tables = new ArrayList<>();
+        this.entityMap = new HashMap<>();
+        this.sqlForeignKeys = new ArrayList<>();
+        this.sqlEnums = new ArrayList<>();
+        this.moduleBuilder = Module.newBuilder("db");
+    }
+
+    public Module introspectDatabase() throws SQLException {
+        ScriptRunner sr = new ScriptRunner(connection);
+        this.tables = sr.getSQLTables(this.getTablesQuery());
+        this.sqlEnums = sr.getSQLEnums(this.getEnumsQuery());
+        tables.forEach(table -> {
+            try {
+                sr.readColumnsOfSQLTable(table, this.getColumnsQuery(table.getTableName()));
+                this.sqlForeignKeys.addAll(sr.readForeignKeysOfSQLTable
+                        (table, this.getForeignKeysQuery(table.getTableName())));
+                sr.readIndexesOfSQLTable(table, this.getIndexesQuery(table.getTableName()));
+            } catch (SQLException e) {
```

Comment:
Rather than catching `SQLException` and throwing it as a `RuntimeException` can't we handle it as `SQLException`?

Reply:
+1. This was resolved in https://github.com/ballerina-platform/persist-tools/pull/330/commits/aed153000919d8a746bfa6a8fc3f76281ea83789